java/fury-core/src/main/java/org/apache/fury/serializer/collection/AbstractMapSerializer.java

```diff
@@ -140,6 +147,378 @@ protected final void writeElements(Fury fury, MemoryBuffer buffer, Map map) {
     }
   }
 
+  protected final void chunkWriteElements(Fury fury, MemoryBuffer buffer, Map map) {
+    Serializer keySerializer = this.keySerializer;
+    Serializer valueSerializer = this.valueSerializer;
+    // clear the elemSerializer to avoid conflict if the nested
+    // serialization has collection field.
+    // TODO use generics for compatible serializer.
+    this.keySerializer = null;
+    this.valueSerializer = null;
+    if (keySerializer != null && valueSerializer != null) {
+      javaChunkWriteWithKVSerializers(buffer, map, keySerializer, valueSerializer);
+    } else if (keySerializer != null) {
+      javaChunkWriteWithKeySerializers(map, buffer, keySerializer);
+    } else if (valueSerializer != null) {
+      javaChunkWriteWithValueSerializers(map, buffer, valueSerializer);
+    } else {
+      genericJavaChunkWrite(fury, buffer, map);
+    }
+  }
+
+  private void javaChunkWriteWithKeySerializers(
```

Comment:
this method is too gig, jvm jit can't inline such big method, could we split it into multiple smaller methods? especially for less frequent code path?